networkx/algorithms/distance_measures.py

```diff
@@ -387,52 +387,72 @@ def diameter(G, e=None, usebounds=False, weight=None):
     return max(e.values())
 
 
-@nx._dispatchable
-def harmonic_diameter(G, sp=None):
+@nx._dispatchable(edge_attrs="weight")
+def harmonic_diameter(G, *, weight=None, sp=None):
     """Returns the harmonic diameter of the graph G.
 
-    The harmonic diameter of a graph is the harmonic mean of the distances
-    between all pairs of distinct vertices. Graphs that are not strongly
-    connected have infinite diameter and mean distance, making such
-    measures not useful. Restricting the diameter or mean distance to
-    finite distances yields paradoxical values (e.g., a perfect match
-    would have diameter one). The harmonic mean handles gracefully
-    infinite distances (e.g., a perfect match has harmonic diameter equal
-    to the number of vertices minus one), making it possible to assign a
-    meaningful value to all graphs.
-
-    Note that in [1] the harmonic diameter is called "connectivity length":
-    however, "harmonic diameter" is a more standard name from the
-    theory of metric spaces. The name "harmonic mean distance" is perhaps
-    a more descriptive name, but is not used in the literature, so we use the
-    name "harmonic diameter" here.
+       The harmonic diameter of a graph is the harmonic mean of the distances
```

Comment:
The changes to indentation level should be undone - they are not stylistically correct and make reviewing much more difficult!